inspirehep/modules/workflows/cli.py

```diff
@@ -75,3 +77,34 @@ def purge(yes_i_know):
     list(current_search.create(ignore_existing=True, index_list=ES_INDICES))
 
     click.secho('Purge completed')
+
+
+@workflows.command()
+@click.argument(
+    'error_message',
+    type=str,
+    metavar='<partial_error_message>'
+)
+@click.option(
+    '--from-beginning',
+    is_flag=True,
+    help="Use this option to restart the workflows from the first step."
+)
+@with_appcontext
+def restart_by_error(error_message, from_beginning):
+    """Restart all the workflows in ERROR matching the given error message."""
+    errors = WorkflowObjectModel.query.filter_by(status=ObjectStatus.ERROR).all()
+    to_restart = [e.id for e in errors if error_message in e.extra_data['_error_msg']]
+
+    click.secho("Found {} workflows to restart from {}".format(
+        len(to_restart),
+        "first step" if from_beginning else "current step"
+    ))
+
+    for wf_id in to_restart:
+        obj = workflow_object_class.get(wf_id)
```

Comment:
You could also change state from error to running or new if started from scratch as otherwise it will be in erorr state until status will be changed which may happen few minutes later.

Reply:
I set `INITIAL` for workflows restarted from the beginning and `RUNNING` for those restarted from current step. :+1: